prov/efa/src/rxr/rxr_rma.c

```diff
@@ -171,11 +173,26 @@ size_t rxr_rma_post_shm_write(struct rxr_ep *rxr_ep, struct rxr_tx_entry *tx_ent
 	msg.msg_iov = tx_entry->iov;
 	msg.iov_count = tx_entry->iov_count;
 	msg.addr = peer->shm_fiaddr;
+	/*
+	 * If CMA is NOT available, FI_MR_VIRT_ADDR will be cleared in shm provider,
+	 * use 0-based offset for shm's RMA operation.
+	 */
+	if (!rxr_env.shm_cma_cap) {
+		for (i = 0; i < tx_entry->rma_iov_count; i++)
+			tx_entry->rma_iov[i].addr = 0;
+	}
 	msg.rma_iov = tx_entry->rma_iov;
 	msg.rma_iov_count = tx_entry->rma_iov_count;
 	msg.context = pkt_entry;
 	msg.data = tx_entry->cq_entry.data;
-	return fi_writemsg(rxr_ep->shm_ep, &msg, tx_entry->fi_flags);
+	ret = fi_writemsg(rxr_ep->shm_ep, &msg, tx_entry->fi_flags);
```

Comment:
We need a discussion here. If shm decided to return -FI_EAGIN, why do we want to queue them?

Reply:
Team reaches an agreement to not add the queuing logic. The current queuing for read will be revisited later.